pkg/tfshim/util/types.go

```diff
@@ -31,6 +31,7 @@ func IsOfTypeMap(tfs shim.Schema) bool {
 
 // CastToTypeObject performs a checked cast from shim.Schema to a TF object (a collection
 // of fields).
+// Note that this is only valid for PF resources - in the TF SDK objects are not represented as maps
```

Comment:
Confusing... Not going to block on this but here's my attempt to update docu https://github.com/pulumi/pulumi-terraform-bridge/pull/2187 on Elem instead around this. I looked at call sites of this function there's 4, mostly look reasonable-ish.

Reply:
https://github.com/pulumi/pulumi-terraform-bridge/pull/2231#discussion_r1687740016

Looks like `CastToTypeObject` does not catch all objects in PF either. I'll raise a PR for this but I wonder how this affects other uses.